### DISCUSSION TITLE
Extending the terrain example with trees and villages

### DISCUSSION THREAD

**Discussion by raypaver** [runevision/LayerProcGen#8] [Q&A] [answered]
First of all: Well done, this looks amazing and runs quite smoothly.

Is it possible to extend the terrain example to include trees (as per you video, even some simple trees)
Could you then also show how you would create a layer for towns/villages, and even if it is simple boxes representing buildings?
I'm not quite grasping how to achieve that.

Thanks.

**Comment by runevision** [comment, marked as answer]
Thanks!

**Trees**  
I excluded trees from the sample scene for a few reasons:

- Because I'm frankly struggling to find a good solution to place lots of trees with good performance in Unity, and the only methods that maybe work require paid assets from the Asset Store.
- Because placing trees is not demonstrating anything specific to the LayerProcGen framework anyway.
- It would make the terrain sample more complicated and bloat the repo with models and (more) textures.

👉 You could look at any other Unity procedural terrain project or tutorial for how to do that and apply the same principle here.

**Towns/villages**  
I haven't generated towns/villages procedurally myself. It's a major undertaking that I can't "just" add to a sample in the framework. 

👉 You would have to look at existing work and discussions about how to do that in general. You could then create a layer with chunks large enough to contain a village/town and generate a village in each chunk, or only in certain chunks, for example. As for how to spawn objects/Prefabs, one of the scenes in the Simple Samples demonstrate spawning spheres.
